File: src/common/updates/UpdateCodes.java
package common.updates;

import common.config.WindowConfig;

public final class UpdateCodes {

    public static final class DefaultCodes {
        public static final String exitToMainGUI = "exitToMainGui";
    }

    public static final class MainMenu {
        public static final String shutdown = "Shutdown";
    }

    public static final class PacMan {
        public static final String gameName = WindowConfig.alexander_title;
        public static final String startGame = gameName + ": Start GameEntryPoint";
        public static final String mainMenu = gameName + ": Mainmenu";
        public static final String showEndScreen = gameName + ":  Show Endscreen";
        public static final String repeatGame = gameName + ": RepeatGame";
    }

    public class Dennis {
	public static final String gameName = WindowConfig.dennis_title;
        public static final String gameReady = "Start Game";
        public static final String gameNotReady = "Don't start";
        public static final String gameLost = "Game lost";
        public static final String gameWon = "Game Won";
        public static final String replay = "replay";
        public static final String continiue = "continiue";
    }

    public static final class TunnelInvader {
        public static final String gameName = WindowConfig.kevin_title;
        public static final String startGame = gameName + " Start GameEntryPoint";
        public static final String playGame = gameName + " Start Game";
        public static final String gameOver = gameName + " Game Over";
        public static final String gameMenu = gameName + " Game Menu";
    }

    public static final class Amir {
        public static final String gameName = WindowConfig.amir_title;
        public static final String startGame = "Start GameEntryPoint";
        public static final String mainMenu = gameName + ": Mainmenu";
        public static final String showEndScreen = gameName + ":  Show Endscreen";
        public static final String repeatGame = gameName + ": RepeatGame";
    }

    public static final class RAM {
        public static final String gameName = WindowConfig.daniel_title;
        public static final String startGame = "Start Game";
        public static final String mainMenu = "MainMenu";
        public static final String fieldSize = "Field size changed";
        public static final String p1Win = "Player 1 wins";
        public static final String p2Win = "Player 2 wins";
        public static final String tie = "It's a tie";
        public static final String quit = "Quit to MainGUI";
    }

    public static final class LK {
        public static final String gameName = WindowConfig.julian_title;
        public static final String startGame = gameName + " Start GameEntryPoint";
        public static final String playGame = gameName + " Start Game";
        public static final String gameOver = gameName + " Game Over";
        public static final String gameMenu = gameName + " Game Menu";
    }

}

File: Common/MainMenu/src/common/updates/UpdateCodes.kt
package common.updates

import common.config.WindowConfig

class UpdateCodes {
    object MainMenu {
        const val shutdown = "Shutdown"
    }

    object PacMan {
        const val gameName = WindowConfig.alexander_title
        const val startGame = "$gameName: Start GameEntryPoint"
        const val mainMenu = "$gameName: Mainmenu"
        const val showEndScreen = "$gameName:  Show Endscreen"
        const val repeatGame = "$gameName: RepeatGame"
    }

    object Dennis {
        const val gameName = WindowConfig.dennis_title
        const val gameReady = "Start Game"
        const val gameNotReady = "Don't start"
        const val gameLost = "Game lost"
        const val gameWon = "Game Won"
        const val replay = "replay"
        const val continiue = "continiue"
    }

    object TunnelInvader {
        const val gameName = WindowConfig.kevin_title
        const val startGame = "$gameName Start GameEntryPoint"
        const val playGame = "$gameName Start Game"
        const val gameOver = "$gameName Game Over"
        const val gameMenu = "$gameName Game Menu"
    }

    object Amir {
        const val gameName = WindowConfig.amir_title
        const val startGame = "Start GameEntryPoint"
        const val mainMenu = "$gameName: Mainmenu"
        const val showEndScreen = "$gameName:  Show Endscreen"
        const val repeatGame = "$gameName: RepeatGame"
    }

    object RAM {
        const val gameName = WindowConfig.daniel_title
        const val startGame = "Start Game"
        const val mainMenu = "MainMenu"
        const val fieldSize = "Field size changed"
        const val p1Win = "Player 1 wins"
        const val p2Win = "Player 2 wins"
        const val tie = "It's a tie"
        const val quit = "Quit to MainGUI"
    }

    object LK {
        const val gameName = WindowConfig.julian_title
        const val startGame = "$gameName Start GameEntryPoint"
        const val playGame = "$gameName Start Game"
        const val gameOver = "$gameName Game Over"
        const val gameMenu = "$gameName Game Menu"
    }
}
